File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/datasourcemodel/Ocgm.java
package uk.gov.justice.digital.hmpps.riskprofiler.datasourcemodel;

import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@Builder
@Data
@EqualsAndHashCode(of = { "nomisId"})
@ToString
public class Ocgm  {
    public static int NOMIS_ID_POSITION = 0;
    public static int OCG_ID_POSITION = 4;
    public static int STANDING_POSITION = 25;

    private String nomisId;
    private String ocgId;
    private String standingWithinOcg;
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/datasourcemodel/Ocgm.kt
package uk.gov.justice.digital.hmpps.riskprofiler.datasourcemodel

data class Ocgm(private val nomisId: String) {
  constructor (nomisId: String, ocgId: String?, standingWithinOcg: String?) : this(nomisId) {
    this.ocgId = ocgId
    this.standingWithinOcg = standingWithinOcg
  }

  var ocgId: String? = null
  var standingWithinOcg: String? = null

  companion object {
    @JvmField
    var NOMIS_ID_POSITION = 0

    @JvmField
    var OCG_ID_POSITION = 4

    @JvmField
    var STANDING_POSITION = 25
  }
}
